### ISSUE TITLE
Selecting a license

### ISSUE THREAD

**Issue by avoelkl** [magento-hackathon/magento2-plugin-visualization#5]
HI @steverobbins @tig-robertgrundeken @KamilBalwierz @beejhuff ,

I've found out that this projects lacks a LICENSE file.
As the five of us are the [contributors](https://github.com/magento-hackathon/magento2-plugin-visualization/graphs/contributors) so far I suggest we agree on a license for this project.

Any recommendations?
My suggestion would be GNU General Public License v3.0 as this one is used for other magento hackathon projects as well.

**Comment by beejhuff** [comment]
That's fine with me, Anna.  We should probably consider a few additional items as part of this effort, so I'm going to suggest we update the title of this issue to something like "Update Repo to Support Best Practices of Open Source Projects & GitHub Community Engagement"

I'm happy to hammer out and submit a few PR's to get feedback / review on all of these in the next few days, but go ahead and PR / Merge in a LICENSE file if you have a GPL v3 one handy.

I'm putting all the items I think should fall under this effort into a checklist we can update as we work through this issue.  We can move this to the main issue description after we've gotten everyone's feedback..

- [ ] **LICENSE.md** - the original item requested when the issue was opened

- [ ]  **README.md** - I've found [this project a good starting point](https://github.com/RichardLitt/standard-readme) for a standard README file template

- [ ] **CONTRIBUTING.md** - just some basic guidelines on how to best contribute to the project, per [GitHub's official documentation](https://help.github.com/articles/setting-guidelines-for-repository-contributors/).

- [ ] **SUPPORT.md** - outlines how we want to support the project.  Mostly used to drive people to IRC / Gitter / Slack channels, but may also include additional info.  
 
   - GitHub's [Help Documentation](https://help.github.com/articles/adding-support-resources-to-your-project/) & [Feature Announcement Blog Post](https://github.com/blog/2400-support-file-support)

   - [Example from the RVM project](https://github.com/rvm/rvm-site/blob/master/content/support.md).

- [ ] **CHANGELOG.md** - this one's probably self-explanatory.  I am a zealot of https://keepachangelog.com and believe in the curation of brief, concise NON-GIT-LOG based change logs, especially if you want newbies & busy devs to participate.  

    - Here's [my ideal example of a GREAT changelog](https://github.com/styled-components/styled-components/blob/master/CHANGELOG.md).  

    - I will stipulate that [this project is a pretty darn good automatic generator of CHANGELOGs](https://github.com/skywinder/github-changelog-generator) and follows the principles of https://keepachangelog.com and is actively developed so may be worthy of consideration for our use.

- [ ] **Templates for Issues & Pull Requests**

    - [ ] _Issue Template_ - this is pretty easy but helps a lot to make sure anyone who submits issues includes those 2-3 items that are important for us to evaluate and respond quickly. Per [GitHub's recommendations.](https://help.github.com/articles/creating-an-issue-template-for-your-repository/)

    - [ ] _Pull Request Template_ - same as above, but for Pull Requests.  Per [GitHub's Recommendations](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/).

    -  I thought [this blog post announcing the issue and pull request templates](https://github.com/blog/2111-issue-and-pull-request-templates) was pretty good at explaining the value it offers and the ease required to set it up...


- [ ] **CODE_OF_CONDUCT.md** - I've noticed some controversy around these (at least in some circles) though I don't understand any of that and this seems to me like something we should do...per [GitHub's Recommendations](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/).


I know that this may seem like a lot of overhead for a relatively simple hackathon project (I'm not arguing against that point), but I'd point out a few things to consider:

  1. I'll take on the responsibility of getting them right and making sure we have a good template to use going forward based on everyone's feedback to remove the burden from you all who did the actual work of coding this up in the first place.

  2.  If we do this well, we can provide these templates to the other hackathon projects and help everyone else who contributes save a ton of time, ensure a standard set of resources that helps more people actively contribute to their projects, and collectively benefit from any of the work we do to enhance these...in fact, I think we can write up a little script that generates standard versions of all of these documents and use that when creating new repositories, though we'd want to chat with @sandermangel or one of the Fabians or who ever is controlling Admin access to the Hackathon org @Schrank @schmengler 

  3. By the way, [here's an example of a project that will automate the creation](https://github.com/gaocegege/maintainer) of most of these standard template files (though it's designed to be run after a repo is manually created, it's not a bad starting point).

**Comment by avoelkl** [comment]
I agree @beejhuff, the LICENSE is just step 1 of some further improvements that need to be done. Thanks for the collection of things that need to be done!
